### DISCUSSION TITLE
Using de.flapdoodle.embed.mongo not only for unittests ?

### DISCUSSION THREAD

**Discussion by fabiengiquel** [flapdoodle-oss/de.flapdoodle.embed.mongo#490] [General]
Hello,
in our software, we need to develop some datas persistence in Java (very small complexity and small volume). The data structure is compatible with document oriented principles. We do not want to manage a separate data server for such a small complexity/volume. So i am looking for embedded document database solutions (and if possible integrated with Spring Boot/Data).

The "de.flapdoodle.embed.mongo" is presented as _"platform neutral way for running mongodb in unittests"_. Beyond unit tests, does it make sense to use "de.flapdoodle.embed.mongo" as a real deployed database, embedded within our java/spring code ?

Thanks in advance for your ideas.
Regards.

**Comment by michaelmosmann** [comment]
@fabiengiquel you can just do that.. it uses the real database.. but you must change the directory where the database files are stored away from temp:) .. but maybe using docker is just easier for that .. (tuning mongodb settings etc.. )

**Reply by fabiengiquel** [reply]
Thanks. And I understand more things after doing some basic tests.
_"_it uses the real database_"_ : ok. I see in my first tests an independant mongod.exe windows process (which is going on even after terminating my java/spring application).
I have changed the db directory, and so i can see the db files. So i was hoping that restarting the java application will reuse the data stored in the db files from last run, but it does not. So the "embedded" mongodb datas needs obviously to be fed at each startup from external source (json...). I understand better why it is oriented to unit tests purpose.
Well, I will consider a more basic Java managed persistence through Jackson framework.
Best regards.

**Reply by michaelmosmann** [reply]
@fabiengiquel if you kill the java process which started a windows process, the windows process is not killed but may run without this parent java process (which is fun and not the case on linux or mac os.. ) .. thats why there is code which tries to stop the mongod process with calling a shutdown command over the network connection.. 

I can create a howto for this kind of situation.. is not that hard.. but maybe its not that obvious. (will create an issue for that)

**Reply by michaelmosmann** [reply]
see https://github.com/flapdoodle-oss/de.flapdoodle.embed.mongo/issues/492

**Reply by michaelmosmann** [reply]
this should work: https://github.com/flapdoodle-oss/de.flapdoodle.embed.mongo/blob/main/docs/UseCases.md#start-mongod-with-persistent-database

**Reply by fabiengiquel** [reply]
Thanks again for your answers.

**Reply by michaelmosmann** [reply]
your welcome:)
